core/modules/mysql/MySqlConfig.cc

```diff
@@ -93,6 +90,28 @@ std::ostream& operator<<(std::ostream &out, MySqlConfig const& mysqlConfig) {
     return out;
 }
 
+void MySqlConfig::checkValidity() const {
+    bool hasError = false;
+    std::string errorMsg = "Invalid MySQL configuration: [";
```

Comment:
Most of the time I suspect that configuration will be valid, in that case you create strings and modify strings but then throw them away which is not very efficient.